R/baselinenowcast.R

```diff
@@ -150,3 +151,310 @@ baselinenowcast.reporting_triangle <- function(
+    nowcast_unit <- colnames(data)[!colnames(data) %in%
+      c(reference_date, report_date, count)]
+  }
+
+  # Split dataframe into a list of dataframes for each nowcast unit
+  if (length(nowcast_unit) != 0) {
+    list_of_dfs <- split(data, interaction(data[nowcast_unit],
+      sep = "___",
+      drop = TRUE
+    ))
+  } else {
+    list_of_dfs <- list(data)
+  }
+
+  # Apply strata sharing if specified
+  if (!is.null(strata_sharing)) {
+    assert_subset(strata_sharing,
+      choices = c("delay", "uncertainty"),
+      empty.ok = TRUE
+    )
+    pooled_df <- combine_triangle_dfs(
+      data = data,
+      reference_date = reference_date,
+      report_date = report_date,
+      count = count
+    )
+    pooled_triangle <- as_reporting_triangle(pooled_df,
+      max_delay = max_delay,
+      reference_date = reference_date,
+      report_date = report_date,
+      count = count
+    )
+    # Get the training volume for all reporting triangles
+    tv <- allocate_reference_times(
+      reporting_triangle = pooled_triangle$reporting_triangle_matrix,
+      scale_factor = scale_factor,
+      prop_delay = prop_delay
+    )
+    if ("delay" %in% strata_sharing) {
+      # Estimate delay once on pooled data
+      delay_pmf <- estimate_delay(
+        reporting_triangle = pooled_triangle$reporting_triangle_matrix,
+        n = tv$n_history_delay
+      )
+    }
+    if ("uncertainty" %in% strata_sharing) {
+      # Estimate uncertainty once on pooled data
+      trunc_rep_tris <- truncate_triangles(
+        reporting_triangle = pooled_triangle$reporting_triangle_matrix,
+        n = tv$n_retrospective_nowcasts
+      )
+      retro_rep_tris <- construct_triangles(trunc_rep_tris)
+      retro_pt_nowcasts <- fill_triangles(retro_rep_tris,
+        max_delay = max_delay,
+        n = tv$n_history_delay
+      )
+      uncertainty_params <- estimate_uncertainty(
+        point_nowcast_matrices = retro_pt_nowcasts,
+        truncated_reporting_triangles = trunc_rep_tris,
+        retro_reporting_triangles = retro_rep_tris,
+        n = tv$n_retrospective_nowcasts,
+        uncertainty_model = uncertainty_model,
+        ...
+      )
+    }
+  }
+
+  combined_result <- imap(list_of_dfs, function(rep_tri_df, name) {
+    rep_tri <- as_reporting_triangle.data.frame(
+      data = rep_tri_df,
+      max_delay = max_delay,
+      delays_unit = delays_unit,
+      reference_date = reference_date,
+      report_date = report_date,
+      count = count
+    )
+
+    nowcast_df <- baselinenowcast.reporting_triangle(
+      data = rep_tri,
+      scale_factor = 2 * scale_factor,
+      prop_delay = prop_delay,
+      output_type = output_type,
+      draws = draws,
+      uncertainty_model = uncertainty_model,
+      uncertainty_sampler = uncertainty_sampler,
+      delay_pmf = delay_pmf,
+      uncertainty_params = uncertainty_params,
+      ...
+    )
+
+    # Split the name of the element in the last and add as a separate column
```

Comment:
This seems like quite a lot of effort? My expectation is you should be able to do this using purr::map_df etc?

Reply:
Also not clear why this happens inside the function?

Reply:
because you are re creating the column names you need, otherwise you'd lose the info within each iteration. Can try using purrr::map_df